src/hotspot/share/compiler/oopMap.cpp

```diff
@@ -191,7 +191,9 @@ void OopMapSet::add_gc_map(int pc_offset, OopMap *map ) {
 }
 
 static void add_derived_oop(oop* base, oop* derived, OopClosure* oop_fn) {
+#if COMPILER2_OR_JVMCI
   DerivedPointerTable::add(derived, base);
+#endif // COMPILER2_OR_JVMCI
```

Comment:
This looks correct and actually reverses the JDK-8253180 change. It is correct because `DerivedPointerTable` is protected by the same `#if`:

```
#if COMPILER2_OR_JVMCI
class DerivedPointerTable : public AllStatic {
...
```